morpheus-graphql-core/src/Data/Morpheus/Types/Internal/AST/Error.hs

```diff
@@ -52,6 +56,17 @@ isInternal :: GQLError -> Bool
 isInternal GQLError {errorType = Just Internal} = True
 isInternal _ = False
 
+custom :: GQLError -> Text -> GQLError
+custom x customError = x {errorType = Just (Custom customError)}
+
+isCustom :: GQLError -> Bool
+isCustom GQLError {errorType = Just (Custom _)} = True
+isCustom _ = False
+
+getCustomErrorType :: GQLError -> Maybe Text
```

Comment:
I added this as well, as for this to be useful there needs to be a way to extract the custom error type from the `GQLError`.